src/ORM/DataList.php

```diff
@@ -487,9 +487,16 @@ public function filterByCallback($callback)
         }
         /** @var ArrayList $output */
         $output = ArrayList::create();
+        $limit = $this->dataQuery->query()->getLimit();
+        $count = 0;
         foreach ($this as $item) {
             if (call_user_func($callback, $item, $this)) {
                 $output->push($item);
+                $count++;
+            }
```

Comment:
I'd add the check for the count even before the callback is run so that (edge-case) limit 0 does nothing at all.

```suggestion
        foreach ($this as $item) {
            if (($count < $limit) && call_user_func($callback, $item, $this)) {
                $output->push($item);
                $count++;
            }
```

Reply:
Yup, good point. I think I'll just move my existing check to be before the existing if statement though - means we only have one condition to handle respecting the limit, and I think the `>=` check is slightly easier to visually parse at a glance than the `<` check.